src/wallet/components/WalletModal.tsx

```diff
@@ -0,0 +1,296 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useConnect, useConnectors } from 'wagmi';
+import { coinbaseWallet } from 'wagmi/connectors';
+import { closeSvg } from '../../internal/svg/closeSvg';
+import { coinbaseWalletSvg } from '../../internal/svg/coinbaseWalletSvg';
+import { defaultAvatarSVG } from '../../internal/svg/defaultAvatarSVG';
+import { walletConnectSvg } from '../../internal/svg/walletConnectSvg';
+import {
+  background,
+  border,
+  cn,
+  color,
+  line,
+  pressable,
+  text,
+} from '../../styles/theme';
+import { useOnchainKit } from '../../useOnchainKit';
+
+type WalletModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  className?: string;
+  onError?: (error: Error) => void;
+};
+
+export function WalletModal({
+  isOpen,
+  onClose,
+  className,
+  onError,
+}: WalletModalProps) {
+  const modalRef = useRef<HTMLDivElement>(null);
+  const [shouldRender, setShouldRender] = useState(isOpen);
+  const { connect } = useConnect();
+  const connectors = useConnectors();
+  const { config } = useOnchainKit();
+
+  useEffect(() => {
+    if (isOpen) {
+      setShouldRender(true);
+    }
+  }, [isOpen]);
+
+  const handleAnimationEnd = () => {
+    if (!isOpen) {
+      setShouldRender(false);
+    }
+  };
+
+  const appLogo = config?.appearance?.logo;
+  const appName = config?.appearance?.name;
+  const privacyPolicyUrl = config?.wallet?.privacyUrl ?? undefined;
+  const termsOfServiceUrl = config?.wallet?.termsUrl ?? undefined;
+
+  const handleCoinbaseWalletConnection = useCallback(() => {
+    try {
+      const cbConnector = coinbaseWallet({
+        preference: 'all',
+      });
+
+      connect({ connector: cbConnector });
+      onClose();
+    } catch (error) {
+      console.error('Coinbase Wallet connection error:', error);
+      if (onError) {
+        onError(
+          error instanceof Error
+            ? error
+            : new Error('Failed to connect wallet'),
+        );
+      }
+    }
+  }, [connect, onClose, onError]);
+
+  const handleWalletConnectConnector = useCallback(() => {
+    try {
+      const walletConnectConnector = connectors.find(
+        (c) => c.type === 'walletConnect',
+      );
+      if (!walletConnectConnector) {
+        console.error('WalletConnect connector not configured');
+        return;
+      }
+      connect({ connector: walletConnectConnector });
+      onClose();
+    } catch (error) {
+      console.error('WalletConnect connection error:', error);
+    }
+  }, [connect, connectors, onClose]);
+
+  const handleKeyDown = useCallback(
+    (e: React.KeyboardEvent | KeyboardEvent) => {
+      if (['Escape', 'Enter', ' '].includes(e.key)) {
```

Comment:
We should remove enter from this - if you press "Enter" while focused on one of the buttons in the modal, then the modal is closed instead of the button being pressed